specs/DesignDiscussions/InputValidation.md

```diff
@@ -0,0 +1,260 @@
+
+As the discussion above shows, depending on the card scenario, it may be difficult for us to determine which fields the card author wants to validate. Any of the above approaches may be correct for *some* cards, but none are appropriate for all cards. In order to address this, the best approach is to allow the card author to specify which inputs are associated with a given action. We will implement this via a new `inputs` property on Action types. The inputs listed will be the ones that are validated, and also the ones that are returned on Submit.
+
+The first option available for inputs is "All." This will validate and submit all inputs on the card. This is the current behavior, and would be the default for submit actions.
+```json
+{
+	"type": "Action.Submit",
+	"title": "Submit",
+	"inputs": "All",
+}
+```
+
+The second option would be "None". This would be useful for, for example, a "Cancel" button where validation is not required, and would be the default for non-Submit (or Invoke action being discussed for Universal Actions) action types (see further discussion of validation for different action types below).
+```json
+{
+	"type": "Action.Submit",
+	"title": "Cancel",
+	"inputs": "None",
+}
+```
+For authors that need more granular control of which inputs to validate, we would also allow them to specify a list of inputs, such as the following:
+```json
+{
+	"type": "Action.Submit",
+	"title": "Cancel",
+	"inputs": ["input1Id", "input2Id"] 
+}
+```
+In order to allow authors to group their inputs, we could also accept container ids. This allows the card author to functionally define a "form" and indicate that the submit button relates to that form:
+```json
+{
+	"type": "Action.Submit",
+	"title": "Cancel",
+	"inputs": ["input1Id", "input2Id", "formContainerId"] 
+}
+```
+
+> Questions:
+>
+> - Do we want to support a "CurrentContainer" or "CurrentContainerAndParent" type option? I think no. Allowing authors to specify containers allows for this behavior, and introducing an option with possibly non-obvious logic in the edge cases may be more confusing than helpful to card authors.
+>
+> - Do we want a typed "Form" container? The design described above allows card authors to use containers as "forms" by specifying them as the inputs for a submit action. Is it helpful to have an explicit "FormContainer" type to use for this? There doesn't seem to be an explicit use for it today. One scenario may be if we have "Form" type properties in the future that would be useful to put on a typed element.
+> - I called the property `"inputs"` rather than something validation specific (`validationInputs`?) because in the submit case it indicates not only the inputs that will be validated, but also those that will be submitted. Are there scenarios in which a submit button would need to validate a different set of inputs than those that it wants to send? I would think **no**, as sending un-validated inputs seems like it could cause problems. 
```

Comment:
>I called the property `"inputs"` [](start = 4, length = 32)

Discussion: On naming, Matt was concerned calling it `inputs` might imply to card authors that input types should go in there. Maybe include "Id" in the name? "AssociatedInputIds"?

Reply:
renamed to associatedInputIds

---
In reply to: [367156518](https://github.com/microsoft/AdaptiveCards/pull/3663#discussion_r367156518) [](ancestors = 367156518)